staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
```

Comment:
Ok, it seems i cannot eliminate the double-registrations yet. Because client-go/pkg/api/v1/helpers.go and reousrce_helper.go still depend on api.Scheme, and because that code is copied instead of being authoratative, i cannot rewrite it to client-go/kubernetes/scheme.Scheme.

With my other refactoring, client-go won't have those two files. After those refactoring are merged, i'll try again to eliminate the double registration.